ShipMonitor/ShipMonitor.cs

```diff
@@ -1183,6 +1198,22 @@ private bool inFighterOrBuggy(string model)
             } while (shipId != profileId);
             Logging.Debug("Current Ship Id is: " + shipId + ", Profile Ship Id is " + profileId);
         }
+
+        static void RaiseOnUIThread(EventHandler handler, object sender)
+        {
+            if (handler != null)
+            {
+                SynchronizationContext uiSyncContext = SynchronizationContext.Current ?? new SynchronizationContext();
+                if (uiSyncContext == null)
```

Comment:
Unless I'm missing something, I don't see how uiSyncContext can be null here.

Reply:
You may be right. I doubt it ever would be.

Reply:
I'm not going to mess with it.